__parse_git_branch.fish

```diff
@@ -1,4 +1,4 @@
 function __parse_git_branch -d "Parse current Git branch name"
   command git symbolic-ref --short HEAD ^/dev/null;
-    or command git show-ref --head -s --abbrev | head -n1 ^/dev/null
+    or string split '\n' (command git show-ref --head -s --abbrev)[1]
```

Comment:
1. Specifying `HEAD` will usually generate less output:
   (removed `-s` just for the example)
    ```shell
    $ command git show-ref --head --abbrev HEAD                                                                                                                                              
    016ea34 HEAD
    228ef9a refs/remotes/origin/HEAD
    ```
    vs:
    ```shell
    command git show-ref --head --abbrev                                                                                                                                                           
    016ea34 HEAD
    284f5b5 refs/heads/master
    cf7af4a refs/remotes/oranja/doctor-db
    016ea34 refs/remotes/oranja/master
    228ef9a refs/remotes/origin/HEAD
    228ef9a refs/remotes/origin/master
    b9c88c6 refs/remotes/upstream/better-contributing
    8c7d4a8 refs/remotes/upstream/chore/bump-license
    74938b0 refs/remotes/upstream/docs/packages
    99d4a3c refs/remotes/upstream/fix-completions
    9eb496d refs/remotes/upstream/fix-double-key-bindings-load
    152cd53 refs/remotes/upstream/fix-git-branch-name
    f5e2ae6 refs/remotes/upstream/fix-line-escape
    d1578c5 refs/remotes/upstream/fix-update-blanks
    f868fb6 refs/remotes/upstream/independent-db
    00b6931 refs/remotes/upstream/installer-refactor
    284f5b5 refs/remotes/upstream/master
    70a32d5 refs/remotes/upstream/omf-install-tests
    72701e3 refs/remotes/upstream/plugins
    a5cd674 refs/remotes/upstream/registry/add-android-sdk
    2c91943 refs/remotes/upstream/registry/add-archlinux
    deb3841 refs/remotes/upstream/registry/add-gityaw
    81edc06 refs/remotes/upstream/registry/add-mou
    6172eac refs/remotes/upstream/registry/add-node-binpath
    4a48ac2 refs/remotes/upstream/registry/add-nodenv
    56313f5 refs/remotes/upstream/registry/add-proxy
    16b7b1d refs/remotes/upstream/registry/add-pyenv
    fb7f4eb refs/remotes/upstream/release-versions
    8217eac refs/remotes/upstream/wip/require-theme
    06aea3a refs/remotes/upstream/wip/require-theme-no-fs
    15e9af8 refs/tags/v2
    ec2b34a refs/tags/v3
    016ea34 refs/tags/v4
    ```

2. It's a little surprising, but the subshell already returns each line as a separate element in the array, so it can be simplified to:
    ```shell
    echo (command git show-ref --head -s --abbrev HEAD)[1]
   ```
   (prints `016ea34`)